tests/Test/Common.hs

```diff
@@ -164,3 +177,61 @@ worldTProperty
 worldTProperty side what = ioProperty $ do
     (prop, _) <- AVL.runOnEmptyCache $ World.evalWorldT def side what
     return prop
+
+-- | Create public key from seed
+mkPubKey :: Char -> PublicKey
+mkPubKey seed = fst (mkKeyPair seed)
+
+-- | Create private key from seed
+mkPrivKey :: Char -> SecretKey
+mkPrivKey seed = snd (mkKeyPair seed)
+
+-- | Create key pair from seed
+mkKeyPair :: Char -> (PublicKey, SecretKey)
+mkKeyPair seed =
+  let (CryptoPassed x) = Ed25519.secretKey (C.replicate 32 seed)
```

Comment:
Why do you use `Char` as a key and replicate it to get 32-bit bytestring, why not use a `ByteString` as a seed directly instead?

Reply:
Argument to `secretKey` need to be 32 bytes long for pattern match to succeed (will get `CryptoFailed` otherwise). Or do you mean something else?

Reply:
I meant, why not just pass a `ByteString` insead of `Char` to `mkKeyPair` and check if it's 32 bytes long instead of accepting only `Chars` and constructing a bytestring of 32 equal bytes, which effectively reduces available seed entropy 2<sup>248</sup> times? 
This isn't by any means critical, as it's only used for tests, but I just didn't quite get the idea here.

Reply:
The idea was just to have a simple way to create deterministic key in pure code and to be sure creation always succeeds. It could of course been done in many other ways. One better way would probable be to take `ByteString` and pad it with zeros to length 32 (or shrink it if longer than 32), as you suggested. It doesn't make much difference IMO except for keyspace. I can update the code to use second approach, no problem.